File: experimental/ImPlay/Implay.Core/PathHelper.cs
namespace ImPlay.Core;

public static class PathHelper
{
    public static string GetConfigDir()
    {
        var exeDir = AppDomain.CurrentDomain.BaseDirectory;
        var portableDir = Path.Combine(exeDir, "portable_config");
        
        if (Directory.Exists(portableDir))
        {
            return portableDir;
        }

        var appData = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
        var path = Path.Combine(appData, "ImPlay");
        Directory.CreateDirectory(path);
        return path;
    }
}

File: experimental/implayfsharpavalonia/Core/PathHelper.fs
namespace ImPlay.Core.Helpers

open System
open System.IO

module PathHelper =
    let GetConfigDir() =
        let exeDir = AppDomain.CurrentDomain.BaseDirectory
        let portableDir = Path.Combine(exeDir, "portable_config")
        
        if Directory.Exists(portableDir) then
            portableDir
        else
            let appData = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)
            let path = Path.Combine(appData, "ImPlay")
            if not (Directory.Exists(path)) then
                Directory.CreateDirectory(path) |> ignore
            path
